backend/controller/controller.go

```diff
@@ -1152,3 +1143,43 @@ func validateCallBody(body []byte, verb *schema.Verb, sch *schema.Schema) error
 	}
 	return nil
 }
+
+// eventExtractor calculates controller events from changes to the state.
+func eventExtractor(diff tuple.Pair[state.State, state.State]) []state.ControllerEvent {
+	var events []state.ControllerEvent
+
+	previous := diff.A
+	current := diff.B
+
+	previousAll := previous.GetDeployments()
+	for _, deployment := range current.GetDeployments() {
+		pd, ok := previousAll[deployment.Key.String()]
+		if !ok {
+			events = append(events, &state.DeploymentCreatedEvent{
+				Module:    deployment.Module,
+				Key:       deployment.Key,
+				CreatedAt: deployment.CreatedAt,
+				Schema:    deployment.Schema,
+				Language:  deployment.Language,
+			})
+		} else if !pd.Schema.Equals(deployment.Schema) {
+			events = append(events, &state.DeploymentSchemaUpdatedEvent{
+				Key:    deployment.Key,
+				Schema: deployment.Schema,
+			})
+		}
+	}
+
+	currentActive := current.GetActiveDeployments()
+	currentAll := current.GetDeployments()
+	for _, deployment := range previous.GetActiveDeployments() {
```

Comment:
I can't quite tell just by looking at this code if it's correct, but it doesnt seem to do what I'd expect.
I'd expect ModuleRemoved to be true if there are no deployments for the module in the new state, but it only seems to be checking if `deployment.Key` is there, so it would be true if a new deployment has replaced it.

Reply:
Yes, that is a bug! Fixing!